src/App.tsx

```diff
@@ -15,48 +17,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABETIC,
+  LENGTH,
+}
+
+type OrderOptions = {
+  sortField: SortType,
+  isReversed: boolean;
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: OrderOptions,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ALPHABETIC:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const readyGoods = getPreparedGoods(goodsFromServer,
+    { sortField, isReversed });
+
+  const handleReverse = () => setIsReversed(!isReversed);
```

Comment:
It is better to toggle the previous value using callback props
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state